lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -597,10 +638,21 @@ void LowerLayersPass::runOnOperation() {
   llvm::sys::SmartMutex<true> mutex;
   circuitMutex = &mutex;
 
+  // Extract information from circuit-level annotations.
+  for (auto anno : AnnotationSet(circuitOp)) {
+    if (!anno.isClass(testBenchDirAnnoClass))
+      continue;
+    auto dir = anno.getMember<StringAttr>("dirname");
+    assert(dir && "invalid test bench annotation");
+    testBenchDir = dir.getValue();
```

Comment:
Last annotation wins works for me.

Reply:
There's a bunch of scattered opinion on this type of thing throughout the passes. There is a stalled effort on making the annotations more first-class. The incompletely rolled out stop-gap is to have dedicated annotation "extractors" which get the information out of a might-exist annotation, while also checking that the annotation is correct. Cf. https://github.com/llvm/circt/blob/10562214dcd5a54724e2e5d50c2d980ef98103b3/lib/Dialect/FIRRTL/FIRRTLAnnotations.cpp#L693

There's a missing function/piece of infrastructure that you tell it what you want to pull out of an array of annotations and it does it and the validation.

All that said, moving towards first-class support where the annotations are actually validated in IR and not when used is much better.